### ISSUE TITLE
Add tests for CSI handlers

### ISSUE THREAD

**Issue by Tyriar** [xtermjs/xterm.js#2117]
- [x] @
- [x] A
- [x] B
- [x] C
- [x] D
- [x] E
- [x] F
- [x] G
- [x] H
- [x] I
- [ ] J
- [ ] K
- [ ] L
- [ ] M
- [ ] P
- [ ] S
- [ ] T
- [ ] X
- [ ] Z
- [ ] `
- [ ] a
- [ ] b
- [ ] c
- [ ] d
- [ ] e
- [ ] f
- [ ] g
- [ ] h
- [ ] l
- [ ] m
- [x] n
- [ ] p
- [ ] q
- [ ] r
- [ ] s
- [ ] u

**Comment by jerch** [comment]
@Tyriar Really nice to get those rolling now we have puppeteer. Once they reach a critical coverage, we can even get rid of the terminal tests with those ugly escape sequence test files.

Furthermore we still have many blind spots regarding correct sequence handling (xterm still fails badly in `vttest` for many tests) which can be fixed gradually with a strong test base.

**Comment by Tyriar** [comment]
Yeah it feels good knowing that we won't break DSR again, that caused a bunch of random issues sitting in VS Code for a few days.

I like the escape sequence tests, just the way they're implemented feels a bit messy. Pretty sure many of the failures they exhibit are bugs/missing functionality from xterm.
